base/bucket_gocb.go

```diff
@@ -1337,28 +1377,188 @@ func (bucket CouchbaseBucketGoCB) IsKeyNotFoundError(err error) bool {
 }
 
 func (bucket CouchbaseBucketGoCB) DeleteDDoc(docname string) error {
-	LogPanic("Unimplemented method: DeleteDDoc()")
-	return nil
+
+	manager, err := bucket.getBucketManager()
+	if err != nil {
+		return err
+	}
+
+	return manager.RemoveDesignDocument(docname)
+
 }
 
 func (bucket CouchbaseBucketGoCB) View(ddoc, name string, params map[string]interface{}) (sgbucket.ViewResult, error) {
-	LogPanic("Unimplemented method: View()")
-	return sgbucket.ViewResult{}, nil
+
+	// Block until there is an available concurrent view op, release on function exit
+	bucket.waitForAvailViewOp()
+	defer bucket.releaseViewOp()
+
+	viewResult := sgbucket.ViewResult{}
+	viewResult.Rows = sgbucket.ViewRows{}
+
+	viewQuery := gocb.NewViewQuery(ddoc, name)
+
+	// convert params map to these params
+	applyViewQueryOptions(viewQuery, params)
+
+	goCbViewResult, err := bucket.ExecuteViewQuery(viewQuery)
+	if err != nil {
+		return viewResult, err
+	}
+
+	if goCbViewResult != nil {
+
+		viewResultMetrics, gotTotalRows := goCbViewResult.(gocb.ViewResultMetrics)
+		if gotTotalRows {
+			viewResult.TotalRows = viewResultMetrics.TotalRows()
+		} else {
+			Warn("Unable to type assert goCbViewResult -> gocb.ViewResultMetrics.  The total rows count may be inaccurate.")
```

Comment:
It should never happen, so I changed the check to be less intrusive and removed the fallback handling.  If it somehow does happen (I guess due to an upstream library change), then it will log a warning rather than panic, since I don't think this is a serious enough error to merit crashing the entire process.

Fixed in 9245e5316dd463000e752eb61be902ea48628f33